OMI-8 / OMI-12 Generate Random ID on insuree Creation and add Insuree attachment

mngoe opened the pull request:
Random ID génération on Insuree Creation process

hirensoni913 committed:
OP-1027: modular IMIS: the front end doesn't support portability of benefits. (#44)

* Limit the list by the logged in user location mapping

* General exception changed to ValueError

* Ignore location filter added to search insuree without a location restriction

* TEST: Fixed bug when testing the insuree photo

edarchis committed:
First fix of generateinsurees, policy broken

Commit:
OP-1080: update backend logic for Insuree number validation (#45)

* OP-1080: update backend logic for Insuree number validation

* OP-1080: ensure to validate inusree number when creating and updating

* OP-1080: update tests to work with current validation

* OP-1080: PR fixes

* OP-1080: fix tests to work when validating insuree number

* OP-1080: apply changes from PR comment

tdethier committed:
Fixed generateinsurees

tdethier committed:
Fixed unit tests

edarchis committed:
Merge pull request #46 from openimis/fix_generateinsuree

Fix generateinsuree

hirensoni913 committed:
Family list is now limited according to the logged in user's mapped location

Commit:
OTC-811: Error messages added for insuree code validation. (#47)

delcroip committed:
Update openmis-module-test.yml

dragos-dobre committed:
Finish feature/OTC-894

OTC-894: Families list is not limited to the user's mapped location

mngoe committed:
Merge pull request #6 from openimis/main

Merge openIMIS Main to Fork main

mngoe committed:
Generate ID automatically. Generation from 1 to 999999999 random int

kevel-dev committed:
Ajout des pieces jointes de l'assuré ref#25122

dragos-dobre commented on insuree/migrations/0014_insureeattachment.py:
```diff
@@ -0,0 +1,25 @@
+                ('idAttachment', models.AutoField(primary_key=True, serialize=False)),
+                ('folder', models.CharField(db_column='Folder', max_length=250)),
+                ('name', models.CharField(db_column='FileName', max_length=250)),
+                ('insuree', models.ManyToManyField(to='insuree.Insuree')),
```
Is there a case when an attachment is attached to multiple insurees?

dragos-dobre commented on insuree/models.py:
```diff
@@ -298,6 +298,23 @@ class Meta:
+    )
+    folder = models.CharField(db_column='Folder', max_length=250, null=False)
+    name = models.CharField(db_column='FileName', max_length=250, null=False)
+    insuree = models.ManyToManyField('Insuree')
```
Is there a case when an attachment is attached to multiple insurees?

mngoe replied:
Fix to change cardinality Pushed

dragos-dobre commented:
@mngoe Did you started from develop branch?

kevel-dev committed:
Correction cardinalités

mngoe commented:
> @mngoe Did you started from develop branch?

@dragos-dobre As discussed, Develop branch can be unstable and difficult to customize (as a lot of fixes are done on multiple repository and update from one repository can break other part of the code; we experienced a lot of difficulties during Cheque Santé project to manage such a volatile development environment)

For such a short development (our customs could be finished next week), I prefered to begin work on main-branch.

Commit:
OTC-878: add permission to all gql queries and mutations (#52)

kevel-dev committed:
BE of insuree attachments ref#25255

mngoe committed:
Merge pull request #7 from mngoe/feature-25255

BE of insuree attachments ref#25255

tdethier committed:
Added modulo 10 verification system (#53)

kevel-dev committed:
Géneration chfid ref#25386

kevel-dev committed:
conformite

mngoe committed:
Merge pull request #8 from mngoe/feature-25386

Géneration chfid ref#25386

kevel-dev committed:
Fix generation chfid lors de le création

kevel-dev committed:
Merge pull request #10 from mngoe/feature-25386

Fix generation chfid lors de le création

kevel-dev committed:
champs Groupe organisé d’appartenance ref#25435

Commit:
Signal for filter extension in insuree query (#54)

kevel-dev committed:
Merge pull request #11 from mngoe/feature-25435

champs Groupe organisé d’appartenance ref#25435

lruzicki committed:
Feature/op 1338: handle insuree filtering by empty string (#55)

* OP-1338: handled empty chfId string input when resolving insurees

* OP-1338: allow CLAIM user to get insuree data

* OP-1338: fixed if statement

* OTC-1338: delete irrelevant perms check

---------

Co-authored-by: lruzicki <lruzicki@gmail.com>

edarchis committed:
Add missing insuree photos report (#56)

Making this query in Django ORM is a overly complex because of the union between ClaimItem and ClaimService that are used for joining to Claim and Product

```mermaid
flowchart TD
    subgraph CDetails
        ClaimItem
        ClaimService
    end
    CDetails --> Claim
    CDetails --> Product
    Insuree --> Family
    Family --> Village
    Village --> Ward
    Ward --> District
    District --> Region
    Claim --> HF
    Claim --> Insuree
    Claim --> ClaimAdmin
```

edarchis committed:
Feature/omt 320 pending enrollments (#57)

* Add missing insuree photos report

Making this query in Django ORM is a overly complex because of the union between ClaimItem and ClaimService that are used for joining to Claim and Product

```mermaid
flowchart TD
    subgraph CDetails
        ClaimItem
        ClaimService
    end
    CDetails --> Claim
    CDetails --> Product
    Insuree --> Family
    Family --> Village
    Village --> Ward
    Ward --> District
    District --> Region
    Claim --> HF
    Claim --> Insuree
    Claim --> ClaimAdmin
```

* OMT-320 pending enrollments report

edarchis committed:
OMT-324 add management commands in package (#58)

dragos-dobre committed:
Merge pull request #59 from openimis/develop

MERGING RELEASE branches

kevel-dev committed:
Ajout de la taille colone titre

lruzicki committed:
OP-1360: set managed to true and upadted Django scheme for all models (#60)

Co-authored-by: lruzicki <lruzicki@gmail.com>

delcroip committed:
Merge pull request #61 from openimis/develop

MERGING RELEASE branches

kevel-dev committed:
change taille du champ chf_id

kevel-dev committed:
Fix code insuree lie a une famille ref#25812

kevel-dev committed:
change chf_id max length

kevel-dev committed:
Champ Type de piece obligatroire ref#25970

kevel-dev committed:
Merge pull request #18 from mngoe/feature-25970

Champ Type de piece obligatoire ref#25970

hirensoni913 committed:
release/23.04

MERGING RELEASE branches

kevel-dev committed:
Merge pull request #19 from openimis/main

Main

kevel-dev committed:
Merge latest main

kevel-dev committed:
Ajustement migration

kevel-dev committed:
Ajout champ firstnam et given name arabe ref#26375

kevel-dev committed:
Merge pull request #20 from mngoe/feature-26375

Ajout champ firstname et given name arabe ref#26375

Keveltiah committed:
Save photo en bd egalement

Keveltiah committed:
Ajustement generation code assuree qui se met a jour lors de l'update
